python/runtime/cudaq/algorithms/py_sample.cpp

```diff
@@ -23,11 +23,6 @@
 
 namespace cudaq {
 
-/// @brief Global cache map of OpaqueArguments
```

Comment:
nit: we could also clean up this var in `py_observe.cpp`

```
/// @brief Global cache map of OpaqueArguments
///
/// For asynchronous execution, we need to construct OpaqueArguments
/// outside of the async lambda invocation. If we don't, then we will be
/// using Python types outside of the current GIL context. Bad things happen
/// then.
std::unordered_map<std::size_t, std::unique_ptr<OpaqueArguments>>
    asyncArgsHolder;
```

Reply:
my bad. good catch